internal/handlers/common/getmore.go

```diff
@@ -38,33 +41,83 @@ func GetMore(ctx context.Context, msg *wire.OpMsg, registry *cursor.Registry) (*
 		return nil, err
 	}
 
-	collection, err := GetRequiredParam[string](document, "collection")
+	// TODO: Use ExtractParam https://github.com/FerretDB/FerretDB/issues/2859
+	v, err := document.Get("collection")
 	if err != nil {
-		return nil, err
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrMissingField,
+			"BSON field 'getMore.collection' is missing but a required field",
+			document.Command(),
+		)
+	}
+
+	collection, ok := v.(string)
+	if !ok {
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrTypeMismatch,
+			fmt.Sprintf(
+				"BSON field 'getMore.collection' is the wrong type '%s', expected type 'string'",
+				commonparams.AliasFromType(v),
+			),
+			document.Command(),
+		)
+	}
+
+	if collection == "" {
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrInvalidNamespace,
+			"Collection names cannot be empty",
+			document.Command(),
+		)
 	}
 
 	cursorID, err := GetRequiredParam[int64](document, document.Command())
 	if err != nil {
-		return nil, err
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrTypeMismatch,
+			"BSON field 'getMore.getMore' is the wrong type, expected type 'long'",
+			document.Command(),
+		)
 	}
 
 	// TODO maxTimeMS, comment
 
 	username, _ := conninfo.Get(ctx).Auth()
 
+	// TODO: Use ExtractParam https://github.com/FerretDB/FerretDB/issues/2859
 	cursor := registry.Cursor(username, cursorID)
 	if cursor == nil {
-		return nil, lazyerrors.Errorf("no cursor %d", cursorID)
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrCursorNotFound,
+			fmt.Sprintf("cursor id %d not found", cursorID),
+			document.Command(),
+		)
+	}
+
+	v, _ = document.Get("batchSize")
+	if v == nil || types.Compare(v, int32(0)) == types.Equal {
+		// TODO: Use 16MB batchSize limit https://github.com/FerretDB/FerretDB/issues/2824
+		// Unlimited default batchSize is used for missing batchSize and zero values,
+		// set 250 assuming it is small enough not to crash FerretDB.
+		v = int32(250)
 	}
 
-	// TODO this logic should be tested
-	batchSize, _ := GetOptionalParam(document, "batchSize", cursor.BatchSize)
-	if batchSize < 0 {
-		batchSize = 101
+	batchSize, err := commonparams.GetValidatedNumberParamWithMinValue(document.Command(), "batchSize", v, 0)
+	if err != nil {
+		return nil, err
 	}
 
 	if cursor.DB != db || cursor.Collection != collection {
-		return nil, lazyerrors.Errorf("cursor %d is for %s.%s, not %s.%s", cursorID, cursor.DB, cursor.Collection, db, collection)
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrUnauthorized,
+			fmt.Sprintf("Requested getMore on namespace '%s.%s', but cursor belongs to a different namespace %s.%s",
+				db,
```

Comment:
```suggestion
			fmt.Sprintf(
				"Requested getMore on namespace '%s.%s', but cursor belongs to a different namespace %s.%s",
				db,
```

Reply:
Thanks :pray: Let me do that on my current aggregation cursor/getMore PR